### PR TITLE
Add `csch` and `sech` operators

### PR DESCRIPTION
**Pull request by Weissnix4711**
Adds the following maths operators:
- sech
- csch
- arsinh
- arcosh
- artanh

And the calc functions:
- asinh
- acosh
- atanh

### CODE REVIEW

**Comment by Enivex** [comment]
Do we really need all of these as built-in? Most are very rarely used, and can easily be added by users.

(Note also that the most common notation is csc and csch for cosec and cosech)

**Comment by laurmaedje** [comment]
Prior art: https://github.com/typst/typst/pull/1090

**Comment by Weissnix4711** [comment]
> Do we really need all of these as built-in? Most are very rarely used, and can easily be added by users.

I don't know, I guess that's up to the maintainers. I think the exposing the inverse functions in calc is useful, but the other math operators I'm not sure about. I guess that decision is up to the maintainers.

>  notation is csc and csch for cosec and cosech

I am aware of the shortened notation, and was initially gonna add those too, until I figured it might be a bit _too_ much. Prolly still is, I don't know.

**Comment by Jollywatt** [comment]
In general, I think it’s better not to have too many trivial math operators defined, because you can just do
```typ
#let csc = math.op("csc")
```
and we want users to realise that. We don’t want users to expect everything to be in the math namespace and then view it as a missing feature when something is missing.

Besides, the more operators we add, the more conventions we inevitably violate. (For example, I have never seen `arsech`, but have seen `arcsech`, etc…)

**Comment by Weissnix4711** [comment]
> We don’t want users to expect everything to be in the math namespace and then view it as a missing feature when something is missing.

That makes sense.

How about shortening the list to this:
- csc (we already have sec and cot, so just makes sense)
- csch (same reasoning)
- arsinh
- arcosh
- artanh

Also, I figured it is prolly a better idea to prefer the abbreviated form of cosec(h). Seems that shorter names are generally preferred here anyways. And I'd think `csc` is probably more common than `cosec`. Though I have nothing to back that up.

> For example, I have never seen arsech, but have seen arcsech, etc

Fyi, they mean exactly the same thing. In normal trig functions, the argument represents the arclength, hence `arc-`, but inverse hyperbolic functions instead represent area. I believe it may also be a regional thing and some countries have different conventions.

**Comment by Jollywatt** [comment]
> * arsinh
> * arcosh
> * artanh

I’d object to those, since Typst already has `math.arcsin, math.arccos, math.arctan`. How about:

* `arcsinh`
* `arccosh`
* `arctanh`

P.S., in the current version of the web app (e546dea2379999a934a6d402a7c56ae4478c3114) it looks like `math.csc` already exists.

**Comment by Enivex** [comment]
> > * arsinh
> > * arcosh
> > * artanh
> 
> I’d object to those, since Typst already has `math.arcsin, math.arccos, math.arctan`. How about:
> 
> * `arcsinh`
> * `arccosh`
> * `arctanh`
> 
> P.S., in the current version of the web app (e546dea2379999a934a6d402a7c56ae4478c3114) it looks like `math.csc` already exists.

The correct names for hyperbolic inverses indeed have ar (for area) and not arc (for arcus). So if definitely **should not** be arcsinh

**Comment by Weissnix4711** [comment]
> math.csc already exists

So it does, thanks. Not sure how I missed that. 

Pushing new changes now..

File: crates/typst-library/src/compute/calc.rs [hunk 1]
```diff
@@ -361,6 +364,57 @@ pub fn tanh(
...
+/// ```example
+/// #calc.asinh(0) \
+/// #calc.asinh(1.5)
+/// ```
+#[func(title = "Inverse Hyperbolic Sine")]
+pub fn asinh(
+    /// The number whose arsinh to calculate.
+    value: Num,
+) -> Angle {
+    // value.asinh()
```
Comment (laurmaedje) [inline crates/typst-library/src/compute/calc.rs:378]:
```suggestion
```

**Review by laurmaedje — COMMENTED** [review]
In general, I don't want to add too many built-in operators, but at least these are fairly specific and there is little potential for collisions across fields. @Enivex what is your opinion regarding the currently remaining set (calc.asinh, calc.acosh, calc.atanh plus the math operators arcosh, arsinh, artanh, csch, and sech)?

**Comment by Enivex** [comment]
> In general, I don't want to add too many built-in operators, but at least these are fairly specific and there is little potential for collisions across fields. @Enivex what is your opinion regarding the currently remaining set (calc.asinh, calc.acosh, calc.atanh plus the math operators arcosh, arsinh, artanh, csch, and sech)?

csch and sech are fairly uncontroversial I'd say.

I have more mixed feelings about the inverse hyperbolic functions, as they're used fairly rarely in my experience. Plus they can easily be expressed in terms of ln() for computations. So I guess it boils down to how "complete" the set of built-in operators should be. I honestly don't know what the answer should be.

**Comment by Dherse** [comment]
I agree with @Enivex, I think having hyperbolic functions makes sense, but the others can be derived from them and other existing functions, as for the inverse functions, I would say they're not needed.

**Comment by laurmaedje** [comment]
Ok, let's add just `csch` and `sech` then.

**Comment by laurmaedje** [comment]
I also hadn't noticed before that the existing inverse trigonometry function in `calc` and the text operators use a different naming convention. That inconsistency within the language is very unfortunate imo. Maybe we should change the calc ones?

**Comment by Enivex** [comment]
> I also hadn't noticed before that the existing inverse trigonometry function in `calc` and the text operators use a different naming convention. That inconsistency within the language is very unfortunate imo. Should we change anything here? Or would a change lead to too much inconsistency with other programming languages?

The use of `a` instead of `ar` (hyperbolic) and `arc` (trigonometric) seems pretty ingrained in programming languages. The question is if that's reason enough to keep those names. Typst is in a somewhat unusual position because it also has the math text operators.

**Comment by laurmaedje** [comment]
If we can swap the order of atan2, I guess we can also rename them.

**Comment by Weissnix4711** [comment]
I don't think it makes any sense to switch away from the convention of `a` for calc and `ar` for text operators. The `a` prefix is following the same convention as in std (and as already pointed out many other languages), but I have never seen someone write `asin`, `acos`, `asinh`, etc in a purely mathematical (not programming) context.

**Comment by laurmaedje** [comment]
Thanks!

**Comment by laurmaedje** [comment]
> I have more mixed feelings about the inverse hyperbolic functions, as they're used fairly rarely in my experience. Plus they can easily be expressed in terms of ln() for computations. So I guess it boils down to how "complete" the set of built-in operators should be. I honestly don't know what the answer should be.

> I agree with @Enivex, I think having hyperbolic functions makes sense, but the others can be derived from them and other existing functions, as for the inverse functions, I would say they're not needed.

Based on these two opinions, I decided to exclude them. I'm sorry that this trimmed the PR down so much.

**Pull request merged**